pandas/core/groupby/ops.py

```diff
@@ -750,7 +746,7 @@ def _aggregate_series_pure_python(self, obj: Series, func: F):
             result[label] = res
 
         result = lib.maybe_convert_objects(result, try_float=0)
-        result = maybe_cast_result(result, obj, numeric_only=True)
+        # TODO: cast to EA once _from_sequence is reliably strict GH#38254
```

Comment:
Inside `maybe_cast_result`, we already have a special case check for not casting back when having datetime or categorical data. You could add periods to that check, and then this line can be left as is, while still fixing the period issue. 
Then we can add the period test, without needing to change the decimal tests.

Reply:
> Inside maybe_cast_result, we already have a special case check for not casting back when having datetime or categorical data. 

That's actually what I find _most_ objectionable about maybe_cast_result.  It is a kludge that special-cases pandas-internal EAs.

Reply:
But as I said below, it would at least be a *consistent* "kludge"... ;) 
Not calling the method here that was exactly written to be called here, it not making it any less kludgy IMO. 

So I think it would be good to add the period test case here, and then focus on #38315 to properly fix it.

Reply:
> not making it any less kludgy

This is the only remaining usage of that kludge, so getting rid of it would certainly make it less kludgy.

> and then focus on #38315 to properly fix it.

It's going to be a while before 1.3, so I'm fine sticking a pin in this to see if a better solution is implemented between now and then.  If it doesn't, we should do this for 1.3.